### PR TITLE
Device namespace fixes

### PR DESCRIPTION
**Pull request by ddutt**
## Description

Three somewhat related bug fixes:
* namespace table: user query was applied too early resulting in either query failures or the wrong data being filtered
* namespace table: the schema didn't contain the lastUpdate column
* device: the filters such as os, vendor etc didn't work correctly because of an incorrect merge with the poller table. There's still a corner case where it fails. 

## Type of change

These are bug fixes. One test was updated to handle the changes caused by these fixes.

## New Behavior

namespace show honors the user filters plus query-str correctly.

## Proposed Release Note Entry

* namespace's schema reflects the actual namespace columns
* namespace honors the filters the user specifies
* device honors the os/vendor/model filters the user specifies.

- [X] I have read the comments and followed the [CONTRIBUTING.md](https://github.com/netenglabs/suzieq/blob/master/CONTRIBUTING.md).
- [X] I have explained my PR according to the information in the comments or in a linked issue.
- [X] My PR source branch is created from the `develop` branch.
- [X] My PR targets the `develop` branch.
- [X] All my commits have `--signoff` applied

### CODE REVIEW

File: suzieq/engines/pandas/device.py [hunk 1]
```diff
@@ -77,7 +80,7 @@ def get(self, **kwargs):
                 .reset_index(drop=True)
 
             df = df.merge(poller_df, on=['namespace', 'hostname'],
-                          how='outer', suffixes=['', '_y'])  \
+                          how='outer', suffixes=['', '_y'], indicator=True) \
```
Comment (LucaNicosia) [inline suzieq/engines/pandas/device.py:83]: I think you added the indicator only for debug. It's not used in the next code lines. We should remove it.

Since you are working on these lines, can you fix the pyright problem as follows?
``` suggestion
                          how='outer', suffixes=('', '_y')) \
```

Comment (ddutt) [reply suzieq/engines/pandas/device.py]: Done
